SalesforceCommerceCloud/pwa-kit
# Enable wishlist functionalities for Einstein recommended products

# Description



_Note: this work is built on top of the `wishlist_rework` branch, for easier PR review, the base branch for this PR is temporarily pointing to `wishlist_rework`. (after `wishlist_rework` is merged, this will point back to `develop`)_

Enable wishlist functionalities for Einstein recommended products.

<img width="1578" alt="Screen Shot 2021-09-21 at 4 01 47 PM" src="https://user-images.githubusercontent.com/10948652/134259151-7b06f859-71e5-4db3-9033-abd91b39a8e1.png">


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Rename `wishlist` icon to `heart` icon to better represent the actual svg icon
-  Decouple `ProductTile` from `ProductScroller` so it's more easier to customize the tiles in the scroller
- Update `RecommendedProducts` component to have wishlist functionalities
- Re work `ProductTile` component, update the API of this component to be clean and easy to use. 
  - decouple the concept of "wishlist" from product tile, instead use "favourite" (make this is an isolated React component for re-usability)

OLD product tile API:
```jsx
<ProductTile
  isWishlistLoading={wishlistLoading.includes(
    productId
  )}
  productSearchItem={productSearchItem}
  onAddToWishlistClick={() =>
    addItemToWishlist(productSearchItem)
  }
  onRemoveWishlistClick={() => {
    removeItemFromWishlist(productSearchItem)
  }}
  isInWishlist={isInWishlist}
/>
```

NEW api:
```jsx
<ProductTile
  product={product}
  enableFavourite={wishlist.isInitialized}
  isFavourite={isInWishlist}
  onFavouriteToggle={(isFavourite) => {
    if (isFavourite) {
      addItemToWishlist(productSearchItem)
    } else {
      removeItemFromWishlist(productSearchItem)
    }
  }}
/>
```

# How to Test-Drive This PR

- `npm start` and go to http://localhost:3000/en-GB/product/25686364M
- ensure you can add/remove einstein recommend products to your wishlist!

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
